app/lib/helpers.rb

```diff
@@ -366,7 +397,96 @@ def slug_contains_invalid(string)
     elsif string.include? "\\"
       retval = true
     end
+  end
+
+  def logged_in_user_is_group_admin_or_site_admin()
+    people = People[cookies[:people]]
+    if people.nil?
+      return false
+    end
+    if people.admin
+      return true
+    end
+    if people.groups().length == 0
+      return false
+    end
+    return true
+  end
 
+  def logged_in_user_is_manager_of_event(event)
+    people = People[cookies[:people]]
+    people.groups().each{ |group|
+      logged_in_users_events = Event.where("group_id = ?", group.id).all
+      logged_in_users_events.each { |logged_in_user_event|
+        if logged_in_user_event.id == event.id
+          return true
+        end
+      }
+    }
+    return false
+  end
+
+  def logged_in_user_is_manager_of_group(group)
+    people = People[cookies[:people]]
+    if people.nil?
+      return false
+    end
+    return people.groups().any?{ |persons_group| persons_group.id == group.id }
+  end
+
+  def update_group_admins_for_person(person)
+    if person.admin == true
+      groups_to_admin = Group.all - person.groups
+      groups_to_admin.each { |group|
+        group.add_person(person)
+      }
+    end
+  end
+
+  #traverse down the tree returning all the events on the way
+  def get_child_events_for_event(event)
+    all_events = []
+    unless event.nil? || event.id.nil?
+      child_events = Event.where("approved = true AND parent_id = ?", event.id).all
+      while child_events.length != 0
+        child_event = child_events.shift
+        child_events += Event.where("approved = true AND parent_id = ?", child_event.id).all
```

Comment:
I don't see what source you're referencing? Are you talking in the context of `get_child_events_for_event(event)`?